utility/src/main/java/org/oppia/android/util/locale/AndroidLocaleProfile.kt

```diff
@@ -3,50 +3,156 @@ package org.oppia.android.util.locale
 import org.oppia.android.app.model.LanguageSupportDefinition.LanguageId
 import org.oppia.android.app.model.RegionSupportDefinition
 import java.util.Locale
+import javax.inject.Inject
 
 /**
  * A profile to represent an Android [Locale] object which can be used to easily compare different
  * locales (based on the properties the app cares about), or reconstruct a [Locale] object.
  *
- * @property languageCode the IETF BCP 47 or ISO 639-2/3 language code
- * @property regionCode the IETF BCP 47 or ISO 3166 alpha-2 region code
+ * Subclasses of this sealed class operate on a language code and/or region code. The language code
+ * is an IETF BCP 47 or ISO 639-2/3 language code, or empty if unknown or not specified. The region
+ * code is an IETF BCP 47 or ISO 3166 alpha-2 region code, or empty if unknown or not specified.
+ *
+ * New instances should be created using [Factory].
```

Comment:
TODO: Document this + why the profile changes were done in the first place.

Included moving Locale creation to the profile (in addition to derivation) to simplify Locale creation for the locale factory class (since there are several different ways to "correct" create the Locale, plus there's now support for producing the root locale).

Reply:
Updated PR description.